### PR TITLE
Feature/add possibility to change role

### PR DESCRIPTION
**Pull request by Leeyaoi**

### CODE REVIEW

File: src/components/ChangeRoleModal/ChangeRoleModal.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import { useGlobalStore } from "../../state/GlobalStore";
+
+interface Props {
+  open: boolean;
+  setOpen: (o: boolean) => void;
+  roleObj: RoleType;
+}
+
+const ChangeRoleModal = ({ open, setOpen, roleObj }: Props) => {
+  const [role, setRole] = useState("");
```
Comment (v12code) [inline src/components/ChangeRoleModal/ChangeRoleModal.tsx]: useState<UserRoles | "">

File: src/components/ChangeRoleModal/ChangeRoleModal.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  const { changeRole } = useGlobalStore();
+
+  const handleClose = () => setOpen(false);
+
+  const handleChange = (event: SelectChangeEvent) => {
+    setRole(event.target.value as string);
```
Comment (v12code) [inline src/components/ChangeRoleModal/ChangeRoleModal.tsx]: setRole(event.target.value as UserRoles);

File: src/components/ChangeRoleModal/ChangeRoleModal.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  };
+
+  const getRole = () => {
```
Comment (v12code) [inline src/components/ChangeRoleModal/ChangeRoleModal.tsx]: Remove it

File: src/components/ChangeRoleModal/ChangeRoleModal.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        return UserRoles.Member;
+      case "1":
+        return UserRoles.Owner;
+      default:
+        return UserRoles.Admin;
+    }
+  };
+
+  const handleClick = () => {
+    roleObj.role = getRole();
```
Comment (v12code) [inline src/components/ChangeRoleModal/ChangeRoleModal.tsx]: roleObj.role = role as UserRoles;

File: src/components/ChangeRoleModal/ChangeRoleModal.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      <div className="ChangeRoleModal">
+        <Typography id="ChangeRoleModal_Label" variant="h1">
+          Select user's role
+        </Typography>
+        <Select
+          id="ChangeRoleModal-select"
+          value={role}
+          onChange={handleChange}
+        >
+          <MenuItem value="0">Member</MenuItem>
```
Comment (v12code) [inline src/components/ChangeRoleModal/ChangeRoleModal.tsx]: Here you can use static values
MenuItem value={UserRoles.Member}>Member</MenuItem

Or just .map

File: src/components/RoleItem/RoleItem.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+interface Props {
+  user: UserType;
+  role: RoleType;
+}
+
+const RoleItem = ({ user, role }: Props) => {
+  const { deleteRole, currentRole, currentUserId } = useGlobalStore();
+  const [open, setOpen] = useState(false);
+
+  const getRoleName = () => {
```
Comment (v12code) [inline src/components/RoleItem/RoleItem.tsx]: Please remove it.
Use this UserRoles[UserRoles.Admin] and so on

File: src/components/RoleItem/RoleItem.tsx [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        src={
+          typeof user !== "undefined" ? user.userPhoto : "../../../profile.jpg"
+        }
+        slotProps={{ img: { referrerPolicy: "no-referrer" } }}
+      />
+      <div className="RoleItem_UserInfo">
+        <Typography variant="h2">{user?.name}</Typography>
+        {getRoleName()}
+      </div>
+      {currentRole !== UserRoles.Member &&
```
Comment (v12code) [inline src/components/RoleItem/RoleItem.tsx]: optional: here you can use same ruleas as in c# - if conditions become too complex, create method with described name such as "isUserCorrect" and use it

**Review by v12code — CHANGES_REQUESTED** [review]

File: src/components/ChangeRoleModal/ChangeRoleModal.tsx (hunk 2 above)
Comment (Leeyaoi) [reply src/components/ChangeRoleModal/ChangeRoleModal.tsx]: image

File: src/components/ChangeRoleModal/ChangeRoleModal.tsx (hunk 1 above)
Comment (Leeyaoi) [reply src/components/ChangeRoleModal/ChangeRoleModal.tsx]: image

**Review by v12code — APPROVED** [review]

**Review by MaliyOna — APPROVED** [review]

**Pull request merged**